From e3d15e7534e2ad40f724ff0b63c8e681aa77de7f Mon Sep 17 00:00:00 2001
From: Radu Pirea <radu-nicolae.pirea@nxp.com>
Date: Wed, 20 Jul 2022 14:47:36 +0300
Subject: [PATCH 010/102] s32cc: fdt: apply lockstep cpu fixup for u-boot dtb

Apply lockstep cpu fixup for the u-boot device tree.

Issue: ALB-9011

Upstream-Status: Pending 

Signed-off-by: Radu Pirea <radu-nicolae.pirea@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/mach-s32/Kconfig      | 1 +
 arch/arm/mach-s32/s32-cc/fdt.c | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/arch/arm/mach-s32/Kconfig b/arch/arm/mach-s32/Kconfig
index 338db238ee..921dbd219b 100644
--- a/arch/arm/mach-s32/Kconfig
+++ b/arch/arm/mach-s32/Kconfig
@@ -63,6 +63,7 @@ config NXP_S32CC
 	select MMC_HS400_ES_SUPPORT
 	select MMC_HS400_SUPPORT
 	select MTD
+	select OF_BOARD_FIXUP
 	select OF_CONTROL
 	select OF_LIVE
 	select OF_SEPARATE
diff --git a/arch/arm/mach-s32/s32-cc/fdt.c b/arch/arm/mach-s32/s32-cc/fdt.c
index 20dcc655f8..a9f744f5d0 100644
--- a/arch/arm/mach-s32/s32-cc/fdt.c
+++ b/arch/arm/mach-s32/s32-cc/fdt.c
@@ -326,3 +326,8 @@ int ft_system_setup(void *blob, bd_t *bd)
 exit:
 	return ret;
 }
+
+int board_fix_fdt(void *rw_fdt_blob)
+{
+	return ft_fixup_cpu(rw_fdt_blob);
+}
-- 
2.17.1

